### DISCUSSION TITLE
Email e Telefono cliente non passati in fattura

### DISCUSSION THREAD

**Discussion by SupervisoreOP** [fattureincloud/api#226] [Q&A] [answered]
Buongiorno,
vi chiedo gentilmente un supporto:

Sto creando una chiamata API per creare ORDINI/FATTURE

EsempiO:
```
{
"data": {
"type": "invoice",
"amount_net": 0,
"amount_due_discount": 0,
"entity": {
"id": 0,
"name": "NOME",
"tax_code": "CF",
"address_street": "INDIRIZZO",
"address_postal_code": "CAP",
"address_city": "CITTA",
"address_province": "PROVINCIA",
"address_extra": "",
"country": "Italia",
"email": "test@test.it",
"phone": "3317823711"
"certified_email": ""
},
```
Premettendo che non passo nessun ID_ENTITY in quanto non voglio censire il cliente, tutto funziona ma le uniche cose che non appaiono sono EMAIL & PHONE. Perchè?

Ho provato invece a impostare una chiamata precedente alla creazione della fattura, per creare il cliente, passando poi l'ID_ENTITY alla successiva call di creazione documento, e in questo caso mi appare email e fattura.

Leggendo però le linee guida, c'è scritto di inserire l'id_entity solo se il cliente esiste, allora seguo la linea guida:
_If the customer is most likely a one-time customer, you can just insert his details directly in the invoice._

Come posso far apparire nel PDF creato email e phone, del cliente che sto passando (senza doverlo censire)?

Grazie a tutti

**Comment by emajo** [comment, marked as answer]
Ciao @SupervisoreOP, se c'è la necessità di visualizzare i campi email phone e fax nel pdf generato allo stato attuale l'unico modo è quello appunto di creare il cliente in anagrafica e passare l'id successivamente in creazione del documento, ne abbiamo già discusso [qui](https://github.com/fattureincloud/api/discussions/93), capiamo che sarebbe utile poter passare quei campi direttamente in creazione del documento, al momento non so dire se ed eventualmente quando questa feature verrà implementata ma speriamo di avere aggiornamenti il prima possibile.

Grazie,
Emanuele.

**Reply by SupervisoreOP** [reply]
Capisco...ma come mai è stata eliminata la possibilità rispetto alla precedente V1?

Comunque al momento bypasso il problema aggiungendo Mail e Phone nell'extra adress....anche se non è il massimo.

Posso chiedere inoltre come intercetto nella chiamata "Create Client" l'eventuale errore se il cliente è già censito? Se fosse già censito c'è modo di recuperare il suo entity_ID (direttamente dalla call di creazione) ? (mi servirebbe poi per effettuare la call di creazione fattura) ...dalla piattaforma da cui parte la chiamata non sono a conoscenza dell'ID cliente (non posso quindi nemmeno eseguire la GET CLIENT).

Grazie
Saluti

**Reply by valmoz** [reply]
Ciao @SupervisoreOP 
suggeriamo di utilizzare la [ListClients](https://developers.fattureincloud.it/docs/api-v2-migration) con [filtering](https://developers.fattureincloud.it/docs/basics/additional-capabilities/filter-results) per andare ad identificare se il cliente è già stato inserito in anagrafica, oppure (se tecnicamente possibile) di salvare anche lato vostro un riferimento ai clienti già inseriti in anagrafica in precedenza.

Con la ListClients otterrete anche l'ID del cliente che vi sarà necessario per effettuare la GetClient.

Grazie
Mauro

**Reply by valmoz** [reply]
Buongiorno @SupervisoreOP è riuscito a risolvere?
